docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,379 @@
+Depending on the consensus algorithm, the determination of the valid block to
+become the chain head will differ. In a Bitcoin Proof of Work consensus, this
+will be the longest chain, whereas PoET uses the measure of aggregate local
+mean (a measure of the total amount of time spent waiting) to determine the
+valid fork. Consensus algorithms with finality, such as PBFT, will only ever
+produce blocks that extend the current head, this function will select the
+block that extends the chain and never the current head.
+
+The ChainController
+===================
+
+The ChainController is responsible for determining which chain the validator is
+currently on and coordinating any change-of-chain activities that need to
+happen. Currently the only listener for chain updates is the BlockPublisher, so
+that it can build a new candidate block on the new chain head.
+
+The ChainController is designed to be able to handle multiple block validation
+activities simultaneously. For instance, if multiple forks form on the network,
+the ChainController can process blocks from all of the competing forks
+simultaneously. This is advantageous as it allows progress to be made even when
+there are several deep forks competing. The current chain can also be advanced
+while a deep fork is being evaluated. This was implemented for cases that could
+happen if a group of validators lost connectivity with the network and later
+rejoined.
+
+Here is the basic flow of the ChainController as a single block is processed. 
+
+.. image:: ../images/journal_chain_controller.*
+   :width: 80%
+   :align: center
+   :alt: Journal Chain Controller Diagram
+
+When a block arrives the ChainController, creates a BlockValidator and
+dispatches it to a thread pool for execution.  Once the BlockValidator has
+completed, it will callback to the ChainController indicating whether the new
+block should be the chain head. This indication falls into 3 cases:
+
+1. The chain head has been updated since the BlockValidator was created. In
+   this case a new BlockValidator is created and dispatched to redo the fork
+   resolution.
+2. The new Block should become the chain head. In this case the chain head is
+   updated to be the new block.
+3. The new Block should not become the chain head. This could be because the
+   new Block is part of a chain that has an invalid block in it, or it is a
+   member of a shorter or less desirable fork as determined by consensus.
+
+The Chain Controller synchronizes chain head updates such that only one
+BlockValidator result can be processed at a time. This is to prevent the race
+condition of multiple fork resolution processes attempting to update the chain
+head at the same time.
+
+Chain Head Update
+-----------------
+
+When the chain needs to be updated, the Chain controller does an update of the
+ChainHead using the BlockStore, providing it with the list of commit blocks
+that are in the new fork and a list of decommit blocks that are in the
```

Comment:
Corrected the last part of this which I think creates a confusion of how things read.  It's now 
"commit blocks that are in the new fork and a list of decommig blocks...old fork, _which_ must be deleted"